### ISSUE TITLE
`func-returns-value` false positive for union type?

### ISSUE THREAD

**Issue by asottile** [python/mypy#19433]
`func-returns-value` seems to have an allowance for functions which themselves return `None` -- but not for functions which return a union with None

**Bug Report**

```python
def returns_none() -> None: ...

def also_returns_none() -> None:
    return returns_none()  # ok

def also_returns_none_sometimes() -> int | None:
    return returns_none()  # E: func-returns-value
```

**To Reproduce**

(see above)

**Expected Behavior**

(no errors)

**Actual Behavior**

```console
$ mypy t.py
t.py:7: error: "returns_none" does not return a value (it only ever returns None)  [func-returns-value]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.16.1
- Mypy command-line flags: n/a
- Mypy configuration options from `mypy.ini` (and other config files): n/a
- Python version used: 3.12.3

**Comment by zzJinux** [comment]
I wonder if this shares the root cause:

```python
def returns_none() -> None: ...

def also_returns_none() -> None:
    return returns_none()  # ok

def returns_tuple() -> tuple[int, None]:
    return 0, returns_none()  # E: func-returns-value
```
